### DISCUSSION TITLE
pymilvus how to delete an ip

### DISCUSSION THREAD

**Discussion by zhenzi0322** [milvus-io/milvus#19170] [Q&A and General discussion]
pymilvus how to delete an ip

**Comment by yhmo** [comment]
Not sure about the "delete an ip".
If it is "delete an entity", please refer this doc: https://milvus.io/docs/v2.1.x/delete_data.md

**Reply by zhenzi0322** [reply]
I am getting the following exception when using delete.

> ValueError: [TypeError("'MutationResult' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]

![image](https://user-images.githubusercontent.com/34839719/189838919-b1eba6d9-c10e-42f8-bdfe-56918e2d9cf3.png)


And query will be able to query it normally.

**Comment by yhmo** [comment]
Is the primary key's name "id"? Is the primary key int64 type?

**Reply by zhenzi0322** [reply]
![image](https://user-images.githubusercontent.com/34839719/189848879-592a2faa-31a7-4b37-a850-b11041722e73.png)

yes.

**Reply by zhenzi0322** [reply]
I solved it now

**Reply by zhenzi0322** [reply]
I have no way to achieve the de-duplication effect by customizing the primary key.

I loaded 10 different data and primary keys into the collection three times and the number in the collection became 30. Normally it should be 10.

**Reply by zhenzi0322** [reply]
What is the cause of this?

**Reply by yhmo** [reply]
Milvus doesn't check the duplication id of each entity.
Let's say you insert 10 entities with id [0, 1, 2, 3, 4, 5,6, 7, 8, 9], the entities number becomes 10.
Then you insert 10 entities with the same id [0, 1, 2, 3, 4, 5,6, 7, 8, 9], the entities number becomes 20.
If you call search(), there could be duplicated results, because the search engine searches all the 20 entities.
If you call query(expr="id in [2]"), milvus only return one result(it just find the first item meet the condition and return).

**Reply by zhenzi0322** [reply]
So how can I get my primary key to be repeatedly inserted without duplication in the collection?

**Reply by yhmo** [reply]
I think milvus cannot achieve that. If a primary key was repeatedly inserted, there must be duplication in the collection.

You can ask milvus to generate unique primary keys for you.
In the following script, I create a collection with a primary key which is auto-generated. When I call the insert(), I don't need input primary keys, the insert() method will return auto-generated ids for me.

```
import random
from pymilvus import (
    connections,
    utility,
    FieldSchema, CollectionSchema, DataType,
    Collection,
)

connections.connect("default", host="localhost", port="19530")

dim = 128
fields = [
    FieldSchema(name="pk", dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=dim)
]
schema = CollectionSchema(fields)
coll = Collection("test", schema)

num = 100
for i in range(10):
    entities = [
        [[random.random() for _ in range(dim)] for _ in range(num)],
    ]
    insert_result = coll.insert(entities)
    print(insert_result.primary_keys)
```

**Reply by zhenzi0322** [reply]
Thank you very much, but I can't use the auto-generated primary keys at the moment, I need to use my own primary keys and images on my side to do so.
